lib/Fmt_ast.ml

```diff
@@ -3757,6 +3757,23 @@ and fmt_module_type c ({ast= mty; _} as xmty) =
       { empty with
         bdy= fmt_longident_loc c lid
       ; epi= Some (fmt_attributes c pmty_attributes ~pre:(Break (1, 0))) }
+  | Pmty_strengthen (mty, lid) ->
+      let {pro; psp; bdy; esp; epi; opn= _; cls= _} =
+        fmt_module_type c (sub_mty ~ctx mty)
+      in
+      { empty with
+        pro=
+          Option.map pro ~f:(fun pro ->
+              open_hvbox 0 $ fmt_if parens "(" $ pro )
```

Comment:
There's code here to put parens around strengthened module types sometimes, but no test that exercises it.  Can you add one?  It looks like trailing attributes might hit it?

Reply:
So parenthesisation of `S with module M = T` seems [completely broken](https://github.com/ocaml-ppx/ocamlformat/issues/2416) and I thought `S with M` should be basically the same so I didn't bother. I now fixed parens for trailing attributes in `S with M` at least (didn't touch `with module`) and added a test.

Reply:
Fair enough!